### PR TITLE
Add a link to JAX Toolbox containers

### PR DESCRIPTION
**Pull request by nouiz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -465,7 +465,8 @@ install [CUDA](https://developer.nvidia.com/cuda-downloads) and
 
 JAX provides pre-built CUDA-compatible wheels for **Linux x86_64 only**. Other
 combinations of operating system and architecture are possible, but require
-[building from source](https://jax.readthedocs.io/en/latest/developer.html#building-from-source).
+[building from source](https://jax.readthedocs.io/en/latest/developer.html#building-from-source)
+or [JAX Toolbox bleeding edge containers](https://github.com/NVIDIA/JAX-Toolbox).
```
Comment (yashk2810) [inline README.md:469]: Maybe we can create a separate section for this?

SHould this be tested before recommending to users?

Comment (nouiz) [reply README.md]: I amended the commit to have a new section.

I also specify what kind of containers they are.
We have a nightly CI in JAX-Toolbox, so they are tested. But they are clearly bleeding edge container and hopefully this is clear in the new version.

File: README.md [hunk 2]
```diff
@@ -391,6 +391,8 @@ installed as the `jaxlib` package. Use the following instructions to install a
 binary package with `pip` or `conda`, or to [build JAX from
```
Comment (hawkinsp) [inline README.md:391]: Work the sentence below in here:

Use the following instructions to install a binary package with `pip` or `conda`, to use a `Docker` container, or to ...

File: README.md [hunk 3]
```diff
@@ -583,6 +585,12 @@ See the `conda-forge`
 [jax](https://github.com/conda-forge/jax-feedstock#installing-jax) repositories
 for more details.
 
+### Containers
```
Comment (hawkinsp) [inline README.md:588]: Can we:
a) move this just below the other CUDA GPU subsections so all the CUDA pieces are together?
b) reword it. Perhaps:

```
### Docker containers: NVIDIA GPU

NVIDIA provides the [JAX Toolbox](...) containers, which are bleeding edge containers containing nightly releases of ...
```

**Review by hawkinsp — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (nouiz) [reply README.md]: Done. I hope I moved where you wanted, not 100% sure.

File: README.md [hunk 4]
```diff
@@ -388,9 +388,11 @@ Some standouts:
 
 JAX is written in pure Python, but it depends on XLA, which needs to be
 installed as the `jaxlib` package. Use the following instructions to install a
-binary package with `pip` or `conda`, or to [build JAX from
+binary package with `pip` or `conda`, to use a
+[Docker containers](#docker-containers-nvidia-gpu), or to [build JAX from
```
Comment (hawkinsp) [inline README.md:392]: Nit: "a Docker container" (singular)

**Review by hawkinsp — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (nouiz) [reply README.md]: Done.

**Review by hawkinsp — APPROVED** [review]

**Pull request merged**